src/Service/Search/SolrSelect.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Search\Service\Search;
+
+use Atoolo\Search\Dto\Search\Query\Facet\FacetField;
+use Atoolo\Search\Dto\Search\Query\Facet\FacetMultiQuery;
+use Atoolo\Search\Dto\Search\Query\Facet\FacetQuery;
+use Atoolo\Search\Dto\Search\Query\Filter\Filter;
+use Atoolo\Search\Dto\Search\Query\QueryOperator;
+use Atoolo\Search\Dto\Search\Query\SelectQuery;
+use Atoolo\Search\Dto\Search\Query\Sort\Criteria;
+use Atoolo\Search\Dto\Search\Query\Sort\Date;
+use Atoolo\Search\Dto\Search\Query\Sort\Headline;
+use Atoolo\Search\Dto\Search\Query\Sort\Name;
+use Atoolo\Search\Dto\Search\Query\Sort\Natural;
+use Atoolo\Search\Dto\Search\Query\Sort\Score;
+use Atoolo\Search\Dto\Search\Result\Facet;
+use Atoolo\Search\Dto\Search\Result\FacetGroup;
+use Atoolo\Search\Dto\Search\Result\SearchResult;
+use Atoolo\Search\SelectSearcher;
+use Atoolo\Search\Service\SolrClientFactory;
+use InvalidArgumentException;
+use Solarium\Component\Facet\Field;
+use Solarium\Core\Client\Client;
+use Solarium\QueryType\Select\Query\Query as SolrSelectQuery;
+use Solarium\QueryType\Select\Result\Result as SelectResult;
+
+/**
+ * Implementation of the searcher on the basis of a Solr index.
+ */
+class SolrSelect implements SelectSearcher
+{
+    /**
+     * @param iterable<SolrQueryModifier> $solrQueryModifierList
+     */
+    public function __construct(
+        private readonly SolrClientFactory $clientFactory,
+        private readonly iterable $solrQueryModifierList,
+        private readonly SolrResultToResourceResolver $resultToResourceResolver
+    ) {
```

Comment:
are `SolrQueryModifier`s required? if so we should check wether it's empty and if not then the argument should be optional:

```suggestion
    public function __construct(
        private readonly SolrClientFactory $clientFactory,
        private readonly SolrResultToResourceResolver $resultToResourceResolver,
        private readonly iterable $solrQueryModifierList = []
    ) {
```